### PR TITLE
CSP-620 Implemented Outer API for Employer Summary

### PR DESCRIPTION
**Pull request by YuvrajPatil-DFE**

### CODE REVIEW

File: src/EmployerAan/SFA.DAS.EmployerAan/Configuration/AanHubApiConfiguration.cs [hunk 1]
```diff
@@ -1,10 +1,3 @@
...
-namespace SFA.DAS.EmployerAan.Configuration;
-
-[ExcludeFromCodeCoverage]
-public class AanHubApiConfiguration
-{
-    public string Url { get; set; } = null!;
-    public string Identifier { get; set; } = null!;
-}
\ No newline at end of file
+public class AanHubApiConfiguration : InnerApiConfiguration { }
```
Comment (DevenKShah) [inline src/EmployerAan/SFA.DAS.EmployerAan/Configuration/AanHubApiConfiguration.cs:3]: Remove the unnecessary use of base class from shared project. This base class is written for shared clients which we are not using.

File: src/EmployerAan/SFA.DAS.EmployerAan/Configuration/CommitmentsV2ApiConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,6 @@
+﻿using System.Diagnostics.CodeAnalysis;
+
+namespace SFA.DAS.EmployerAan.Configuration;
+
+[ExcludeFromCodeCoverage]
+public class CommitmentsV2ApiConfiguration : InnerApiConfiguration { }
```
Comment (DevenKShah) [inline src/EmployerAan/SFA.DAS.EmployerAan/Configuration/CommitmentsV2ApiConfiguration.cs:6]: Remove the unnecessary use of base class from shared project. This base class is written for shared clients which we are not using.

File: src/EmployerAan/SFA.DAS.EmployerAan.UnitTests/Application/Employer/Queries/GetEmployerMemberSummary/GetEmployerMemberSummaryQueryHandlerTests.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    public class GetEmployerMemberSummaryQueryHandlerTests
+    {
+        [Test, MoqAutoData]
+        public async Task Handle_ReturnEmployerMemberSummary(
+        [Frozen] Mock<ICommitmentsV2ApiClient> apiClient,
+        GetEmployerMemberSummaryQueryHandler handler,
+        GetEmployerMemberSummaryQuery query,
+        AccountsSummary expectedAccountsSummary,
+        ApprenticeshipsFilterValues expectedApprenticeshipsFilterValues,
+        CancellationToken cancellationToken)
```
Comment (DevenKShah) [inline src/EmployerAan/SFA.DAS.EmployerAan.UnitTests/Application/Employer/Queries/GetEmployerMemberSummary/GetEmployerMemberSummaryQueryHandlerTests.cs:21]: Needs indentation here

File: src/EmployerAan/SFA.DAS.EmployerAan/Application/Employer/Queries/GetEmployerMemberSummary/GetEmployerMemberSummaryQueryHandler.cs [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        public async Task<GetEmployerMemberSummaryQueryResult?> Handle(GetEmployerMemberSummaryQuery request, CancellationToken cancellationToken)
+        {
+            var responseEmployerAccount = await _commitmentsV2ApiClient.GetEmployerAccounts(request.EmployerAccountId, cancellationToken);
+            var responseEmployerSummary = await _commitmentsV2ApiClient.GetApprenticeshipsSummaryForEmployer(request.EmployerAccountId, cancellationToken);
+
+            GetEmployerMemberSummaryQueryResult result = new();
+
+            if (responseEmployerAccount.ResponseMessage.StatusCode == System.Net.HttpStatusCode.OK)
+            {
+                result.ActiveCount = responseEmployerAccount.GetContent()!.ApprenticeshipStatusSummaryResponse!.FirstOrDefault()!.ActiveCount;
```
Comment (DevenKShah) [inline src/EmployerAan/SFA.DAS.EmployerAan/Application/Employer/Queries/GetEmployerMemberSummary/GetEmployerMemberSummaryQueryHandler.cs:24]: I am assuming this is a list of all the accounts and their respective apprentice counts. We need count of the account that the user is registered with not the first item in the list.

**Review by DevenKShah — CHANGES_REQUESTED** [review]

**Review by DevenKShah — APPROVED** [review]

**Review by MarkFCain — APPROVED** [review]

**Review by MarkFCain — APPROVED** [review]

**Review by DevenKShah — DISMISSED** [review]

**Review by MarkFCain — DISMISSED** [review]

**Review by DevenKShah — APPROVED** [review]

**Review by MarkFCain — APPROVED** [review]

**Pull request merged**